From 78824cce41327a0f3858e7e2cc53275d29262b53 Mon Sep 17 00:00:00 2001
From: xubing <bing.xu@intel.com>
Date: Tue, 26 Mar 2024 16:34:27 +0800
Subject: [PATCH] Set default value of display hardware settings

Set default value of display hardware settings, include
Hue, Contrast, Saturation, Whitebalance and Luminance.

Test: Feature funciotn is ready, test OK.

Tracked-On: OAM-116761
Signed-off-by: xubing <bing.xu@intel.com>
---
 packages/SettingsProvider/res/values/defaults.xml | 12 +++++++++++-
 .../providers/settings/DatabaseHelper.java        | 15 +++++++++++++++
 .../display/color/DisplayTransformManager.java    |  8 ++++----
 3 files changed, 30 insertions(+), 5 deletions(-)

diff --git a/packages/SettingsProvider/res/values/defaults.xml b/packages/SettingsProvider/res/values/defaults.xml
index 8e6e251ff3f2..652c7fe199df 100644
--- a/packages/SettingsProvider/res/values/defaults.xml
+++ b/packages/SettingsProvider/res/values/defaults.xml
@@ -29,8 +29,18 @@
     <bool name="def_auto_time">true</bool>
     <bool name="def_auto_time_zone">true</bool>
     <bool name="def_accelerometer_rotation">false</bool>
-    <!-- Default screen brightness, from 0 to 255.  102 is 40%. -->
+    <!-- Default screen backlight, from 0 to 255.  102 is 40%. -->
     <integer name="def_screen_brightness">102</integer>
+    <!-- Default screen saturation, from 0 to 100.  50 is 50%. -->
+    <integer name="def_screen_saturation">50</integer>
+    <!-- Default screen hue, from 0 to 360.  180 is 50%. -->
+    <integer name="def_screen_hue">180</integer>
+    <!-- Default screen contrast, from 0 to 255.  128 is 50%. -->
+    <integer name="def_screen_contrast">128</integer>
+    <!-- Default screen luminance,  from 0 to 255.  128 is 50%. -->
+    <integer name="def_screen_luminance">128</integer>
+    <!-- Default screen whitebalance, from 0 to 100.  100 is 100%. -->
+    <integer name="def_screen_whitebalance">100</integer>
     <bool name="def_screen_brightness_automatic_mode">false</bool>
     <fraction name="def_window_animation_scale">100%</fraction>
     <fraction name="def_window_transition_scale">100%</fraction>
diff --git a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
index 268603fa8b0d..7b3afe0bd6b2 100644
--- a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
+++ b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
@@ -2256,6 +2256,21 @@ class DatabaseHelper extends SQLiteOpenHelper {
             loadIntegerSetting(stmt, Settings.System.SCREEN_BRIGHTNESS,
                     R.integer.def_screen_brightness);
 
+            loadIntegerSetting(stmt, Settings.System.SCREEN_CONTRAST,
+                    R.integer.def_screen_contrast);
+
+            loadIntegerSetting(stmt, Settings.System.SCREEN_HUE,
+                    R.integer.def_screen_hue);
+
+            loadIntegerSetting(stmt, Settings.System.SCREEN_LUMINANCE,
+                    R.integer.def_screen_luminance);
+
+            loadIntegerSetting(stmt, Settings.System.SCREEN_SATURATION,
+                    R.integer.def_screen_saturation);
+
+            loadIntegerSetting(stmt, Settings.System.SCREEN_WHITEBALANCE,
+                    R.integer.def_screen_whitebalance);
+
             loadIntegerSetting(stmt, Settings.System.SCREEN_BRIGHTNESS_FOR_VR,
                     com.android.internal.R.integer.config_screenBrightnessForVrSettingDefault);
 
diff --git a/services/core/java/com/android/server/display/color/DisplayTransformManager.java b/services/core/java/com/android/server/display/color/DisplayTransformManager.java
index 4d3223aa727a..8be8db7f8520 100644
--- a/services/core/java/com/android/server/display/color/DisplayTransformManager.java
+++ b/services/core/java/com/android/server/display/color/DisplayTransformManager.java
@@ -56,19 +56,19 @@ public class DisplayTransformManager {
      */
     public static final int LEVEL_COLOR_MATRIX_SATURATION = 150;
     /**
-     * Color transform level used to adjust the color saturation of the display.
+     * Color transform level used to adjust the color hue of the display.
      */
     public static final int LEVEL_COLOR_MATRIX_HUE = 100;
     /**
-     * Color transform level used to adjust the color saturation of the display.
+     * Color transform level used to adjust the color contrast of the display.
      */
     public static final int LEVEL_COLOR_MATRIX_CONTRAST = 100;
     /**
-     * Color transform level used to adjust the color saturation of the display.
+     * Color transform level used to adjust the color white balance of the display.
      */
     public static final int LEVEL_COLOR_MATRIX_WHITEBALANCE = 100;
     /**
-     * Color transform level used to adjust the color saturation of the display.
+     * Color transform level used to adjust the color luminance of the display.
      */
     public static final int LEVEL_COLOR_MATRIX_LUMINANCE = 150;
     /**
-- 
2.34.1

